### ISSUE TITLE
no-useless-escape false positive with tagged template literals

### ISSUE THREAD

**Issue by Kovensky** [eslint/eslint#7383]
Noticed with eslint 3.8.0.

The `no-useless-escape` rule incorrectly flags `\` characters inside `String.raw`-modified template strings. `String.raw` specifically ignores any escaping and emits the backslashes as-is, so they are never escaping anything at all.

Code snippet that triggered the false positives:

```js
this.startTokenMatch = new RegExp(String.raw`^-\*-\* ${this.token} start \{(.+)\} \*-\*-$`)
```

In this case, the backslashes are there specifically so they get emitted as backslashes, as they are escaping regexp metacharacters; but this false positive is unrelated to regexps.

**Comment by not-an-aardvark** [comment]
Confirmed. Thanks for the report.

This is caused by https://github.com/eslint/eslint/pull/7332, which allows `no-useless-escape` to report uselessly-escaped characters in template literals. This is reasonable behavior in most cases, but I think we shouldn't report these characters in tagged template literals, since the tag function can access the raw text.

```js
/* eslint no-useless-escape: 2 */

`foo \.`; // this line should be reported
myFunc`foo \.`; // this line should not be reported
```
